src/common/telemetry/lib/Telemetry.cpp

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "Telemetry.hpp"
+#include "ParameterSets.hpp"
+
+#include <algorithm>
+#include <chrono>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <fstream>
+#include <iostream>
+#include <nlohmann/json.hpp>
+#include <sstream>
+#include <thread>
+
+using namespace MAT;
+
+namespace Telemetry
+{
+
+TelemetryManager::TelemetryManager(bool enableDebug, int teardownTime, OsConfigLogHandle logHandle)
+    : m_log(logHandle)
+    , m_logManager(nullptr)
+    , m_logger(nullptr)
+{
+#if defined(DEBUG) || defined(_DEBUG) || !defined(NDEBUG)
+    SetLoggingLevel(LoggingLevelDebug);
+#endif
+
+    m_logConfig["name"] = "TelemetryModule";
+    m_logConfig["version"] = TELEMETRY_VERSION;
+    m_logConfig["config"]["host"] = "*";
+    m_logConfig[CFG_BOOL_ENABLE_TRACE] = enableDebug;
+    m_logConfig[CFG_INT_TRACE_LEVEL_MIN] = 0;
+    m_logConfig[CFG_INT_MAX_TEARDOWN_TIME] = teardownTime;
+
+    status_t status = STATUS_SUCCESS;
+    m_logManager.reset(LogManagerProvider::CreateLogManager(m_logConfig, status));
+
+    if (STATUS_SUCCESS == status)
+    {
+        OsConfigLogInfo(m_log, "Telemetry initialized successfully.");
+
+        m_logger = m_logManager->GetLogger(API_KEY, "logger_direct");
+        if (!m_logger)
+        {
+            OsConfigLogError(m_log, "Failed to get logger instance");
+        }
+    }
+    else
```

Comment:
Do we need to throw here? CreateLogManager failed, the object is probably on a bad state. Throw here.